File: java/src/main/java/com/fleetmgr/sdk/client/event/output/facade/ChannelsClosing.java
package com.fleetmgr.sdk.client.event.output.facade;

import com.fleetmgr.sdk.client.traffic.Channel;

import java.util.List;

/**
 * Created by: Bartosz Nawrot
 * Date: 18.09.2018
 * Description:
 */
public class ChannelsClosing extends FacadeEvent {

    private List<Channel> channels;

    public ChannelsClosing(List<Channel> channels) {
        super(Type.CHANNELS_CLOSING);
        this.channels = channels;
    }

    public List<Channel> getChannels() {
        return channels;
    }

    @Override
    public String toString() {
        return "ChannelsClosing{" +
                "channels=" + channels +
                '}';
    }
}

File: cpp/source/event/output/ChannelsClosing.cpp
#include "event/output/ChannelsClosing.hpp"

#include <sstream>
#include <iterator>

using namespace fm;
using namespace fm::event::output;

ChannelsClosing::ChannelsClosing(std::shared_ptr<std::vector<traffic::IChannel*>> _channels) :
    FacadeEvent(CHANNELS_CLOSING),
    channels(_channels)
{
}

std::shared_ptr<std::vector<traffic::IChannel*>> ChannelsClosing::getChannels() const
{
    return channels;
}

std::string ChannelsClosing::toString() const
{
    std::ostringstream oss;
    oss << "[";
    for (auto c : *channels)
    {
        oss << c->getId() << ",";
    }
    oss << "]";
    return "CHANNELS_CLOSING: " + oss.str();
}
